besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -865,40 +864,35 @@ static class PrivacyOptionGroup {
   private BesuController besuController;
   private BesuConfigurationImpl pluginCommonConfiguration;
 
-  private BesuComponent besuComponent;
   private final Supplier<ObservableMetricsSystem> metricsSystem =
-      Suppliers.memoize(
-          () ->
-              besuComponent == null || besuComponent.getObservableMetricsSystem() == null
-                  ? MetricsSystemFactory.create(metricsConfiguration())
-                  : besuComponent.getObservableMetricsSystem());
+      Suppliers.memoize(() -> MetricsSystemFactory.create(metricsConfiguration()));
```

Comment:
are we not getting metrics system from dagger anymore?

Reply:
Right, so this was what i meant by a "broad cycle" if BesuCommand depends on BesuComponent, it's gonna eventually depend on all the things, when in reality it just needed the plugin context.

There are four MetricsSystems we know how to create, Prometheus, OTEL and the NoOp and Stub systems used in all the testing. If a client is using a metrics system that is configurable, (Prometheus and OTEL) then Besu command MUST know how to create it, since the details needed come from BesuCommand.

So we never really needed that hack in the first place, and instead can just create a DaggerModule to provide the NoOp or Stub MetricsSystems should we need them in tests.